packages/cloud_firestore/cloud_firestore_platform_interface/lib/src/method_channel/method_channel_document_reference.dart

```diff
@@ -94,42 +94,49 @@ class MethodChannelDocumentReference extends DocumentReferencePlatform {
   }
 
   @override
-  Stream<DocumentSnapshotPlatform> snapshots(
-      {bool includeMetadataChanges = false}) {
+  Stream<DocumentSnapshotPlatform> snapshots({
+    bool includeMetadataChanges = false,
+  }) {
     assert(includeMetadataChanges != null);
-    int handle = MethodChannelFirebaseFirestore.nextMethodChannelHandleId;
-    Completer<void> onListenComplete = Completer<void>();
 
     // It's fine to let the StreamController be garbage collected once all the
     // subscribers have cancelled; this analyzer warning is safe to ignore.
     StreamController<DocumentSnapshotPlatform>
         controller; // ignore: close_sinks
+
+    StreamSubscription<dynamic> snapshotStream;
     controller = StreamController<DocumentSnapshotPlatform>.broadcast(
       onListen: () async {
-        MethodChannelFirebaseFirestore.documentObservers[handle] = controller;
-        await MethodChannelFirebaseFirestore.channel.invokeMethod<void>(
-          'DocumentReference#addSnapshotListener',
+        final observerId = await MethodChannelFirebaseFirestore.channel
+            .invokeMethod<String>('DocumentReference#snapshots');
+        snapshotStream =
+            MethodChannelFirebaseFirestore.documentSnapshotChannel(observerId)
+                .receiveBroadcastStream(
           <String, dynamic>{
-            'handle': handle,
-            'firestore': firestore,
             'reference': this,
+            'firestore': firestore,
```

Comment:
I don't think this is used for document snapshot queries. might be something legacy.